### DISCUSSION TITLE
optimizing yolo5 for detecting objects from recorded video

### DISCUSSION THREAD

**Discussion by AshishMahendra** [ultralytics/yolov5#2645] [Q&A]
I'm using yolo5 for object detection from recorded video, the model is taking 1/3rd time to complete the entire processing,e.g. a 24 hour video is taking 8 hour to complete the processing. How to reduce the processing time further.

**Comment by glenn-jocher** [comment]
@AshishMahendra tips for faster inference are always the same, no surprises:

- Use a smaller model
- Use a smaller image size
- Use larger batch sizes
- Use a faster GPU

**Reply by AshishMahendra** [reply]
I'm using RTX2080Ti GPU

**Comment by Ownmarc** [comment]
In case of video processing, you could skip frames if your use case allows it. You could also process the frames in a separated thread to accumulate a certain number of frames in advance so you are able to keep the gpu busy at all time and send batches of frames instead of 1 by 1. You could also split the video into parts and have multiple gpu process certain part in parallel and then bringing back the detections all together. There is many tricks you can do

**Reply by AshishMahendra** [reply]
I"m splitting the video to 1 hour, using 1 process for frame generation and 1 for tracking the object id and have used 2 processes for object detection, is there anything else i can do?
